.github/workflows/detekt.yml

```diff
@@ -20,12 +18,19 @@ jobs:
         with:
           java-version: 11
           distribution: zulu
-      - uses: burrunan/gradle-cache-action@v1
+      - uses: gradle/gradle-build-action@v2
         with:
           gradle-version: wrapper
-          # additional files to calculate key for dependency cache
-          gradle-dependencies-cache-key: |
-            buildSrc/**/Versions.kt
-      # Workaround for bug in gradle-cache-action + detekt
-      - name: Run gradle command
-        run: ./gradlew detektAll
+          arguments: |
+            detektAll
+      - name: Copy SARIF reports into a single directory
+        if: ${{ always() }}
+        run: |
+          mkdir -p build/detekt-sarif-reports
+          i=0
+          find . -name "detekt.sarif" | while read -r f; do echo "$f -> detekt-$i.sarif" && cp $f build/detekt-sarif-reports/detekt-$i.sarif || echo Cannot copy && echo Copied && i=$((i+1)); done
```

Comment:
May be also print which iteration couldn't be copied, i.e. `i` in `echo Cannot copy`?

Reply:
It's logged right after message with file names, so I think it should be enough as is.